### PR TITLE
Modifies text indentation behaviour

### LINKED ISSUE #14559: Multiple problems with TextEdit selection while indenting
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
Current master: 1401b07

**Issue description:**
Making indentations while having text selected has the following problems:

- [ ] Indenting to the left will remove a character from the start of the selection (this also happens when undoing right indentations):
screenshot at 2017-12-11 13-52-30

- [x] Indenting to the right will add a character from the start of the selection:
screenshot at 2017-12-11 13-58-06

- [ ] Undoing left indentations will remove a character from the end of the selection:
screenshot at 2017-12-11 13-53-54

- [x] There were also times that the indentation options vanished from the context menu, but I'm not sure what causes this.

### PR DESCRIPTION
**Pull request by MattUV**
Partially fixes #14559

### CODE REVIEW

**Comment by YeldhamDev** [comment]
This does not fix the third and fourth points of the issue.

**Comment by MattUV** [comment]
Oh no :( I ran several tests without getting the issues anymore.
I'll investigate more this evening.

**Comment by MattUV** [comment]
I realise I may have misunderstood the term "undoing". I thought you meant "left indentation after a right indentation", while you meant "Ctrl+Z after a right indentation"

**Comment by MattUV** [comment]
Alright, I have dug deeper into the code, and optimized it a little bit. There were some useless
repetitions between `text_edit.cpp`, `screen_text_editor.cpp` and `shader_editor_plugin.cpp` , and the method was easily  modified to work when text is not selected. That's why I also renamed `indent_selection_left()` and `indent_selection_right()` to `indent_left()` and `indent_right()`
I also fixed the 4th issue, by displaying "Indent left" and "Indent right" all the time in the context menu.

However, I was unable to fix the 3rd issue, nor the 2nd part of the 1st issue. I have some troubles with the "undo" part, I can't figure out how to record the selection in TextOperation. Maybe a dedicated issue would be a good thing, as it would keep selections even when undoing a text deletion.

File: scene/gui/text_edit.cpp [hunk 1]
```diff
@@ -1664,17 +1664,22 @@ void TextEdit::backspace_at_cursor() {
...
 
-	if (!is_selection_active()) {
-		return;
-	}
+	int start_line;
+	int end_line;
 	begin_complex_operation();
-	int start_line = get_selection_from_line();
-	int end_line = get_selection_to_line();
+
```
Comment (brainsick) [inline scene/gui/text_edit.cpp]: If I'm reading/interpreting the Travis CI output correctly, it thinks you've violated a code formatting rule.  I believe it's asking you to delete two tab characters.

File: scene/gui/text_edit.cpp [hunk 2]
```diff
@@ -1688,23 +1693,32 @@ void TextEdit::indent_selection_right() {
...
```
Comment (brainsick) [inline scene/gui/text_edit.cpp]: other tab character

**Review by brainsick — COMMENTED** [review]
If I'm interpreting the Travis CI output correctly, you've violated a code formatting rule.  I believe you need to delete two tab characters.

**Comment by MattUV** [comment]
Thank you, that is corrected now.

File: scene/gui/text_edit.cpp [hunk 3]
```diff
@@ -5657,4 +5681,4 @@ TextEdit::TextEdit() {
 }
 
 TextEdit::~TextEdit() {
-}
+}
```
Comment (hpvb) [inline scene/gui/text_edit.cpp]: Very nitpicky but we default to having a final newline in a file.

**Pull request merged**